src/webui/api/appcontroller.cpp

```diff
@@ -1083,6 +1085,38 @@ void AppController::defaultSavePathAction()
     setResult(BitTorrent::Session::instance()->savePath().toString());
 }
 
+QDir::Filters parseDirectoryContentMode(const QString &visibility)
+{
+    if (visibility == u"dirs")
+        return QDir::Dirs;
+    if (visibility == u"files")
+        return QDir::Files;
+    if (visibility == u"all")
+        return (QDir::Dirs | QDir::Files);
+    throw APIError(APIErrorType::BadParams, u"Invalid mode, allowed values: all, dirs, files"_s);
```

Comment:
qbt enable translations for api error messages:
```suggestion
    throw APIError(APIErrorType::BadParams, tr("Invalid mode, allowed values: all, dirs, files"));
```
And to ensure proper scope for `tr()` I suggest you make this helper function a lambda within `getDirectoryContentAction()` if there are no plans to reuse it. Otherwise move it to `AppController` class private function.

Reply:
Hmm... If API error messages need to be translated then my previous [suggestion](https://github.com/qbittorrent/qBittorrent/pull/20314#discussion_r1493251272) should be rejected.

Reply:
Otherwise you cannot use tr() in this helper function

Reply:
@Chocobo1 

> And to ensure proper scope for tr() I suggest you make this helper function a lambda within getDirectoryContentAction() if there are no plans to reuse it. Otherwise move it to AppController class private function.

I don't know what do you mean, I did not work with qt translations and I have no idea how to write lambda which will initialize context.

Reply:
>I don't know what do you mean, I did not work with qt translations and I have no idea how to write lambda which will initialize context.

First, apply the suggested change above. (from the first post of this thread)
Then: 
```c++
void AppController::getDirectoryContentAction()
{
    requireParams({u"dirPath"_s});

    // move the function to here:
    const auto parseDirectoryContentMode = [](const QString &visibility) -> QDir::Filters
    {
        // ...
    };
  ```

Reply:
@Chocobo1 Done